app/nexus.go

```diff
@@ -0,0 +1,545 @@
+								endpointNameFlag,
+							},
+							Action: func(ctx *cli.Context) error {
+								endpointName := ctx.String(endpointNameFlag.Name)
+								endpoint, err := c.getEndpointByName(endpointName)
+								if err != nil {
+									return err
+								}
+								return PrintProto(endpoint)
+							},
+						},
+						{
+							Name:        "list",
+							Aliases:     []string{"l"},
+							Usage:       "List Nexus Endpoints (EXPERIMENTAL)",
+							Description: "This command lists all Nexus Endpoint configurations on the Server",
+							Flags:       []cli.Flag{},
+							Action: func(ctx *cli.Context) error {
+								endpoints, err := c.listEndpoints("")
+								if err != nil {
+									return err
+								}
+								return PrintObj(endpoints)
+							},
+						},
+						{
+							Name:    "create",
+							Aliases: []string{"c"},
+							Usage:   "Create a new Nexus Endpoint (EXPERIMENTAL)",
+							Description: "This command creates a new Nexus Endpoint on the Server.\n" +
+								"An endpoint name is used by in workflow code to invoke Nexus operations.\n" +
+								"The endpoint target is a worker and `--target-namespace` and `--target-task-queue` must both be provided.\n" +
+								"This will fail if an endpoint with the same name is already registered",
+							Flags: []cli.Flag{
+								endpointNameFlag,
+								targetNamespaceFlag,
+								targetTaskQueueFlag,
+								allowNamespaceFlag,
+								RequestIDFlag,
+							},
+							Action: func(ctx *cli.Context) error {
+								resp, err := c.createEndpoint(
+									ctx.String(endpointNameFlag.Name),
+									ctx.String(targetNamespaceFlag.Name),
+									ctx.String(targetTaskQueueFlag.Name),
+									ctx.StringSlice(allowNamespaceFlag.Name),
+									ctx.String(RequestIDFlag.Name),
+								)
+								if err != nil {
+									return err
+								}
+								return PrintProto(resp)
+							},
+						},
+						{
+							Name:    "update",
+							Aliases: []string{"u"},
+							Usage:   "Update an existing Nexus Endpoint (EXPERIMENTAL)",
+							Description: "This command updates an existing Nexus Endpoint on the Server.\n" +
+								"An endpoint name is used by in workflow code to invoke Nexus operations.\n" +
+								"The endpoint target is a worker and `--target-namespace` and `--target-task-queue` must both be provided.\n" +
+								"Only the fields that are provided will be updated",
+							Flags: []cli.Flag{
+								endpointNameFlag,
+								targetNamespaceFlagOptional,
+								targetTaskQueueFlagOptional,
+								ResourceVersionFlag,
+								RequestIDFlag,
+							},
+							Action: func(ctx *cli.Context) error {
+								endpointName := ctx.String(endpointNameFlag.Name)
+								targetNamespaceID := ctx.String(targetNamespaceFlagOptional.Name)
+								targetTaskQueue := ctx.String(targetTaskQueueFlagOptional.Name)
+								resourceVersion := ctx.String(ResourceVersionFlag.Name)
+
+								if targetNamespaceID == "" && targetTaskQueue == "" {
```

Comment:
[Nit] what if they provided updates are identical to the values on the existing endpoint?

Reply:
Added a separate check ✔️